app/js/views/services-list-item.jsx

```diff
@@ -306,6 +314,15 @@ export default class ServicesListItem extends React.Component {
         return this.renderGenericService();
     }
   }
+
+  getServiceName() {
+    // If service has tags, let's use them as service identifier.
+    const nameFromTags = this.service.getTags().reduce((name, tag) => {
+      return name ? `${name}, ${tag}` : tag;
+    }, '');
```

Comment:
Damn, I can't explain why I used this super-complex-and-clumsy solution :D Will definitely remove it!